frontend/src/pages/SignUp/index.tsx

```diff
@@ -98,69 +72,94 @@ const SignUp: React.FC = () => {
           <Stack
             component="form"
             direction="column"
-            spacing={1.5}
             sx={(theme) => ({
               marginTop: theme.spacing(2),
               marginBottom: theme.spacing(2),
             })}
-            onSubmit={handleSignUp}
-            noValidate
+            onSubmit={handleSubmit((data) =>
+              signup(
+                data.firstName,
+                data.lastName,
+                data.username,
+                data.email,
+                data.password
+              )
+            )}
           >
-            <CustomTextField
-              label="First Name"
-              size="small"
```

Comment:
Without this attribute, I feel that the spacing looks a little too huge haha and the screen easily becomes vertically scrollable with the button going beyond the viewport:

![image](https://github.com/user-attachments/assets/72791067-7b70-4c87-a330-c4ed65586ef8)

I thought it might be better to restrict the heights, but what do you think!